### ISSUE TITLE
Adapting TG Models to Edge Regression 

### ISSUE THREAD

**Issue by shenyangHuang** [shenyangHuang/TGB#1]
Training: 
1. change the loss function from binary cross entropy to MSE 
2. For edge weights, it should be used in loss function instead of edge feature
3. change the output MLP of TGN etc. to be a regression MLP
4. Same negative sampling as link prediction, 1 positive edge, 1 negative edge (weight = 0) but predicting weights now 

Test and Validation: 
1. predict for all node pairs and computing the MSE across all predictions, same for test and validation
2. Consider other metrics in the future for the edge weights.
